src/main/java/org/datadog/jmxfetch/util/CustomLogger.java

```diff
@@ -1,123 +1,166 @@
+                if (end == -1 || end <= start || end > str.length()) {
+                    end = str.length();
+                }
+                return str.substring(start + 1, end);
+            }
 
-        String logPattern = logFormatRfc3339 ? LAYOUT_RFC3339 : LAYOUT;
-
-        PatternLayout layout = PatternLayout.newBuilder()
-            .withConfiguration(config)
-            .withPattern(logPattern)
-            .build();
-
-        if (logLocation != null
-                && !ConsoleAppender.Target.SYSTEM_ERR.toString().equals(logLocation)
-                && !SYSTEM_ERR_ALT.equals(logLocation)
-                && !ConsoleAppender.Target.SYSTEM_OUT.toString().equals(logLocation)
-                && !SYSTEM_OUT_ALT.equals(logLocation)) {
-
-            target = "FileLogger";
-
-            RollingFileAppender fa = RollingFileAppender.newBuilder()
-                .setConfiguration(config)
-                .withName(target)
-                .withLayout(layout)
-                .withFileName(logLocation)
-                .withFilePattern(logLocation + ".%d")
-                .withPolicy(SizeBasedTriggeringPolicy.createPolicy("5MB"))
-                .withStrategy(DefaultRolloverStrategy.newBuilder().withMax("1").build())
-                .build();
-
-            fa.start();
-            config.addAppender(fa);
-            ctx.getRootLogger().addAppender(config.getAppender(fa.getName()));
-
-            log.info("File Handler set");
-        } else {
-
-            if (logLocation != null
-                    && (ConsoleAppender.Target.SYSTEM_ERR.toString().equals(logLocation)
-                        || SYSTEM_ERR_ALT.equals(logLocation))) {
-
-                ConsoleAppender console = (ConsoleAppender)config.getAppender("CONSOLE");
-                console.stop();
-                config.getRootLogger().removeAppender("CONSOLE");
-                ctx.updateLoggers();
-
-                ConsoleAppender ca = ConsoleAppender.newBuilder()
-                    .setConfiguration(config)
-                    .withName(logLocation)
-                    .setTarget(ConsoleAppender.Target.SYSTEM_ERR)
-                    .withLayout(layout)
-                    .build();
-
-                ca.start();
-                config.addAppender(ca);
-                ctx.getRootLogger().addAppender(config.getAppender(ca.getName()));
+            @Override
+            public synchronized String format(LogRecord lr) {
+                SimpleDateFormat simpleDateFormat = new SimpleDateFormat(dateFormat,
+                                                        Locale.getDefault());
+                return String.format(format,
+                    simpleDateFormat.format(new Date()).toString(),
+                    // NOTE(remy): these conversions may generate a lot of garbage
+                    //      (jaime): if performance impact over a sustained period of time
+                    //               isn't too bad, we can probably live with this.
```

Comment:
The comment in the code here shouldn't end up getting merged as-is

Reply:
removing it